tests/test_transformations.py

```diff
@@ -0,0 +1,127 @@
+                                               np.array([[[5, 5], [5, 5]],
+                                                         [[11, 1], [0, 5]]])),
+                                              (10, 2, 10, 'last',
+                                               np.array([[[3, 2], [10, 7]],
+                                                         [[8, 3], [9, 0]]]))
+                                              ])
+def test_video_downsampling(input_fps, output_fps, random_seed, strategy,
+                            expected_video, video_fixture):
+    new_video = transformations._downsample_array(full_array=video_fixture,
+                                                  input_fps=input_fps,
+                                                  output_fps=output_fps,
+                                                  strategy=strategy,
+                                                  random_seed=random_seed)
+    assert np.array_equal(expected_video, new_video)
+
+
+@pytest.mark.parametrize(("coordinate_pair, window_size, video_shape,"
+                          "expected_pair"), [
+                        ((0, 9), (10, 10), (512, 512), (4, 9)),
+                        ((9, 0), (10, 10), (512, 512), (9, 4)),
+                        ((511, 4), (10, 10), (512, 512), (506, 4)),
+                        ((4, 511), (10, 10), (512, 512), (4, 506)),
+                        ((5, 5), (3, 3), (5, 5), (3, 3))])
+def test_coordinate_window_shift(coordinate_pair, window_size, video_shape,
+                                 expected_pair):
+    transformed_center = transformations.get_transformed_center(coordinate_pair,
+                                                                     window_size,
+                                                                     video_shape)
+    assert transformed_center == expected_pair
+
+
+def test_mp4_conversion(video_fixture):
+    output_path = Path(__file__).parent
+    output_path = output_path / 'video.mp4'
+
+    fps = 30
+
+    transformations.transform_to_mp4(video=video_fixture,
+                                     output_path=output_path.as_posix(),
+                                     fps=fps),
+
+    frames = []
+    reader = imageio.get_reader(output_path.as_posix(), mode='I', fps=fps,
+                                size=(2, 2), pixelformat="gray")
+    for frame in reader:
+        gray_frame = cv2.cvtColor(frame, cv2.COLOR_RGB2GRAY)
+        frames.append(gray_frame)
+    frames = np.stack(frames)
+
+    # codec for mp4 encoding changes the video slightly between h5 and mp4
+    # and vice versa, so numbers are not exactly the same. Instead general
+    # structure of video array loaded. Cv color conversion does not rectify
+    # this error unfortunately, this has been attempted and small errors
+    # persist
+
+    assert frames.shape == video_fixture.shape
+
+    os.remove(output_path)
+
+
+@pytest.mark.parametrize("video,coord,box_size,expected",
```

Comment:
It's helpful to explain your test cases in the comments when it's not immediately clear.

Reply:
Sounds good